src/test/java/teammates/sqlui/webapi/QueryLogsActionTest.java

```diff
@@ -0,0 +1,404 @@
+package teammates.sqlui.webapi;
+
+import java.time.Instant;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import teammates.common.datatransfer.logs.EmailSentLogDetails;
+import teammates.common.datatransfer.logs.ExceptionLogDetails;
+import teammates.common.datatransfer.logs.FeedbackSessionAuditLogDetails;
+import teammates.common.datatransfer.logs.GeneralLogEntry;
+import teammates.common.datatransfer.logs.LogSeverity;
+import teammates.common.datatransfer.logs.RequestLogDetails;
+import teammates.common.datatransfer.logs.RequestLogUser;
+import teammates.common.datatransfer.logs.SourceLocation;
+import teammates.common.util.Const;
+import teammates.ui.output.GeneralLogsData;
+import teammates.ui.webapi.JsonResult;
+import teammates.ui.webapi.QueryLogsAction;
+
+/**
+ * SUT: {@link QueryLogsAction}.
+ */
+public class QueryLogsActionTest extends BaseActionTest<QueryLogsAction> {
+    private static final String GOOGLE_ID = "user-googleId";
+
+    JsonResult actionOutput;
+
+    long startTimeForFailCases = Instant.now().toEpochMilli();
+    long endTimeForFailCases = startTimeForFailCases - 1000;
+    long endTimeForSuccessCases = Instant.now().toEpochMilli();
+    long startTimeForSuccessCases = endTimeForSuccessCases - 1000 * 60 * 60 * 24;
+    Map<String, Object> requestParams = new HashMap<>();
+    Map<String, Object> requestHeaders = new HashMap<>();
+
+    String severity = "INFO";
+    String infoLogTrace1 = "info log trace 1";
+    String infoLogTrace2 = "info log trace 2";
+    String infoLogInsertId1 = "info log insert id 1";
+    String infoLogInsertId2 = "info log insert id 2";
+    String infoLogTextPayload1 = "info log text palyload 1";
+    String infoLogTextPayload2 = "info log text palyload 2";
+    SourceLocation infoLogSourceLocation1 = new SourceLocation("file1", 1L, "func1");
+    SourceLocation infoLogSourceLocation2 = new SourceLocation("file2", 2L, "func2");
+    long infoLogTimestamp1 = endTimeForSuccessCases - 1000 * 60 - 1;
+    long infoLogTimestamp2 = endTimeForSuccessCases - 1000 * 60 - 2;
+    RequestLogUser infoLogUserInfo1 = new RequestLogUser();
```

Comment:
Since this variable is only used in the `setUp` method, I think we can define it inside there

Reply:
Hmm oki, that's true